### PR TITLE
Simplify find_localzeros

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133325

Instead of doing an N^2 connected thing, only do simplifications for binary max/min, and for very simple situations.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

Differential Revision: [D64135230](https://our.internmc.facebook.com/intern/diff/D64135230)

### CODE REVIEW

**Comment by ezyang** [comment]
Surprising. I'll figure out a simplified fix that fixes the test cases

**Comment by ezyang** [comment]
It looks related to Max simplification:

* Missing constant prop now `Max(-1.0, 0.0) != 0.0`
* Value ranges enabled simplification not working `Eq(3*Max(1, u0), 3*u0) ~> True`, see also https://github.com/pytorch/pytorch/issues/125914

File: torch/utils/_sympy/functions.py [hunk 1]
```diff
@@ -761,49 +762,30 @@ def _find_localzeros(cls, values, **options):
...
-
-        return False
+        if len(values) == 2:
+            ix = 1 if cls is Max else 0
+            if values[0] in (0.0, 0) and values[1].is_nonnegative:
+                return {values[ix]}
+            elif values[1] in (0.0, 0) and values[0].is_nonnegative:
+                return {values[1 - ix]}
+            elif values[1] == 1 and values[0].is_positive:
+                return {values[1 - ix]}
```
Comment (isuruf) [inline torch/utils/_sympy/functions.py:776]: Return 1 in case `values[0].is_nonpositive`?

Comment (ezyang) [reply torch/utils/_sympy/functions.py:776]: Do you think maybe I should just use the old is connected algorithm for the 2-ary case? Maybe that will also be fast.

Comment (isuruf) [reply torch/utils/_sympy/functions.py:776]: I don't think it would be fast. What you have here is pretty fast.

Comment (ezyang) [reply torch/utils/_sympy/functions.py:776]: okey dokey. In that case, I'm going to omit the nonpositive case, we basically never have values like that lol

**Review by albanD — APPROVED** [review]
Sounds good.
Should we have specific tests to ensure the cases we care about are still covered?

**Comment by ezyang** [comment]
> Should we have specific tests to ensure the cases we care about are still covered?

So, all the cases here were implicitly gotten from the test suite (because I deleted this entirely and then things started failing)...

**Comment by ezyang** [comment]
This one is pretty simple: I need to support ternary simplification: `Eq(Max(1, 257, u3), Max(257, u3)`

**Comment by ezyang** [comment]
@ezyang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D64135230).

**Pull request closed without merging**